### ISSUE TITLE
Counter values are returned as 0 if counter > 2^31

### ISSUE THREAD

**Issue by scotloach** [exhuma/puresnmp#76]
puresnmp 1.6.2.post1 on python 3.7.5.

I am polling a system and counters (such as if_mib counters) above 2^31 are being returned as 0 because they are coming back as a signed int and then failing the > 0 check.

Probably the value needs to be masked into an unsigned value before it's checked.

**Comment by exhuma** [comment]
Thanks for reporting this. I'll have a look

**Comment by exhuma** [comment]
Can you run the following code with the appropriate values for the failing OID and add the hexdump to this issue (this will help a ton in resolving this issue):

    import logging
    from puresnmp import get

    logging.basicConfig(level=logging.DEBUG)
    get(IP, COMMUNITY, OID)

**Comment by exhuma** [comment]
nb: The hexdump will help to identify the exact data-type returned by the system (`Counter` or `Counter64`), and give me the problematic values to use in testing.

**Comment by scotloach** [comment]
Here you go:
```DEBUG:puresnmp.transport:Sending packet to 192.168.140.43:161 (attempt 1/3)
30 2d 02 01 01 04 08 43  61 6e 6f 70 79 72 6f a0   0-.....Canopyro.
1e 02 04 5e 40 bb 13 02  01 00 02 01 00 30 10 30   ...^@........0.0
0e 06 0a 2b 06 01 02 01  02 02 01 0a 02 05 00      ...+...........
DEBUG:puresnmp.transport:Received packet:
30 31 02 01 01 04 08 43  61 6e 6f 70 79 72 6f a2   01.....Canopyro.
22 02 04 5e 40 bb 13 02  01 00 02 01 00 30 14 30   "..^@........0.0
12 06 0a 2b 06 01 02 01  02 02 01 0a 02 41 04 84   ...+.........A..
43 20 f8                                           C .
```

**Comment by exhuma** [comment]
Has been merged into the main development branch and will be released soon as 1.7.4
